src/components/molecules/Visualizer/Engine/Cesium/cameraLimiter.ts

```diff
@@ -0,0 +1,237 @@
+
+    const {
+      cameraLimitterTargetWidth: width = targetWidth,
+      cameraLimitterTargetLength: length = targetLength,
+    } = property ?? {};
+
+    const { cartesianArray } = calcBoundaryBox(
+      geodsic,
+      length / 2 + rectangleHalfHeight,
+      width / 2 + rectangleHalfWidth,
+    );
+
+    setCameraViewOuterBoundaries(cartesianArray);
+  }, [cesium, property, geodsic]);
+
+  // Manage camera limiter conditions
+  useEffect(() => {
+    const camera = getCamera(cesium?.current?.cesiumElement);
+    const viewer = cesium?.current?.cesiumElement;
+    if (!viewer || viewer.isDestroyed() || !property?.cameraLimitterEnabled || !limiterDimensions)
+      return;
+    if (camera) {
+      viewer.camera.setView({
+        destination: getAllowedCameraDestination(camera, limiterDimensions),
+        orientation: {
+          heading: viewer.camera.heading,
+          pitch: viewer.camera.pitch,
+          roll: viewer.camera.roll,
+          up: viewer.camera.up,
+        },
+      });
+    }
+  }, [camera, cesium, property, limiterDimensions]);
+
+  return {
+    limiterDimensions,
+    cameraViewOuterBoundaries,
+    cameraViewBoundariesMaterial,
+  };
+}
+
+export const cameraViewBoundariesMaterial = new PolylineDashMaterialProperty({
+  color: Color.RED,
+});
+
+export type InnerLimiterDimensions = {
+  cartographicDimensions: {
+    rightDemention: Cartographic;
+    leftDemention: Cartographic;
+    topDemention: Cartographic;
+    bottomDemention: Cartographic;
+  };
+  cartesianArray: Cartesian3[];
+};
+
+export const getGeodsic = (
+  viewer: CesiumViewer,
+  property: SceneProperty["cameraLimiter"] | undefined,
+): { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic } | undefined => {
+  if (!property || !property?.cameraLimitterTargetArea) return undefined;
+  const ellipsoid = viewer.scene.globe.ellipsoid;
+
+  const centerPoint = Cartesian3.fromDegrees(
+    property.cameraLimitterTargetArea.lng,
+    property.cameraLimitterTargetArea.lat,
+    0,
+  );
+
+  const cartographicCenterPoint = Cartographic.fromCartesian(centerPoint);
+  const normal = ellipsoid.geodeticSurfaceNormal(centerPoint);
+  const east = Cartesian3.normalize(
+    Cartesian3.cross(Cartesian3.UNIT_Z, normal, new Cartesian3()),
+    new Cartesian3(),
+  );
+  const north = Cartesian3.normalize(
+    Cartesian3.cross(normal, east, new Cartesian3()),
+    new Cartesian3(),
+  );
+
+  const geodesicVertical = new EllipsoidGeodesic(
+    cartographicCenterPoint,
+    Cartographic.fromCartesian(north),
+    ellipsoid,
+  );
+  const geodesicHorizontal = new EllipsoidGeodesic(
+    cartographicCenterPoint,
+    Cartographic.fromCartesian(east),
+    ellipsoid,
+  );
+  return { geodesicVertical, geodesicHorizontal };
+};
+
+export const calcBoundaryBox = (
+  geodsic: { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic },
```

Comment:
```suggestion
  geodesic: { geodesicVertical: EllipsoidGeodesic; geodesicHorizontal: EllipsoidGeodesic },
```
!! Don't forget to update wherever this function is being called. !!